misc/dbt-materialize/tests/adapter/test_deploy.py

```diff
@@ -270,6 +270,19 @@ def test_cluster_owner_permissions(self, project):
 
 
 class TestRunWithDeploy:
+    @pytest.fixture(scope="class")
+    def dbt_profile_target(self):
+        return {
+            "type": "materialize",
+            "threads": 1,
+            "host": "{{ env_var('DBT_HOST', 'localhost') }}",
+            "user": "materialize",
+            "pass": "password",
+            "database": "materialize",
+            "port": "{{ env_var('DBT_PORT', 6875) }}",
+            "cluster": "quickstart",
```

Comment:
@morsapaes adding this override here which should fix the two failing tests as we will explicitly have the cluster defined in the `profile.yml`

Reply:
I'm still confused as to why this is failing if _a_ cluster name is passed in the `deployment` configuration. Or, we shouldn't need to pass a default cluster in `profiles.yml`, and if that's different than the cluster specified in the `deployment` configuration, what happens?

```yaml
class TestRunWithDeploy:
    @pytest.fixture(scope="class")
    def project_config_update(self):
        return {
            "vars": {
                "deployment": {
                    "default": {"clusters": ["quickstart"], "schemas": ["public"]}
                }
            }
        }
```

Reply:
The deployment variables are only used in all deploy actions:

https://github.com/MaterializeInc/materialize/blob/2c12a216884a1767446541818bd59b2c7a85da6e/misc/dbt-materialize/dbt/include/materialize/macros/deploy/deploy_init.sql#L19

https://github.com/MaterializeInc/materialize/blob/2c12a216884a1767446541818bd59b2c7a85da6e/misc/dbt-materialize/dbt/include/materialize/macros/deploy/deploy_init.sql#L29

All those actions are responsible for preparing your infrastructure like creating the blue clusters and schemas and then doing the cutover but not the actual dbt runs.

The actual `dbt run` and the `dbt run --var "deploy: True"` do not use those variables but instead rely on the cluster from the `profile.yml` or the cluster configuration itself or in the past the default `quickstart` cluster.

As far as I can see, the reason why you can't just use the deployment variables is that if you have multiple clusters defined in there, there is no way for us to really know which cluster should be used/associated with which object. So all that the `dbt run --var "deploy: True"` does is to add the deploy suffix to the customer pre-configured cluster name.

Reply:
> As far as I can see, the reason why you can't just use the deployment variables is that if you have multiple clusters defined in there, there is no way for us to really know which cluster should be used/associated with which object.

Sorry, I meant when you're processing each cluster passed in the variables, not picking a random cluster! Thanks for clarifying, @bobbyiliev. 🫶